api/migrations/0002_data_migration.py

```diff
@@ -97,6 +99,66 @@ def _write_languages_and_levels(self):
         )
         LanguageAndLevel.objects.bulk_create(language_and_level_objects)
 
+    def _write_enrollment_tests(self):
+        """Writes `EnrollmentTest`, `Question`, and `EnrollmentTestQuestionOption` to database."""
+        EnrollmentTest = self.apps.get_model(APP_NAME, "EnrollmentTest")
+        EnrollmentTestQuestion = self.apps.get_model(APP_NAME, "EnrollmentTestQuestion")
+        EnrollmentTestQuestionOption = self.apps.get_model(
+            APP_NAME, "EnrollmentTestQuestionOption"
+        )
+
+        AgeRange = self.apps.get_model(APP_NAME, "AgeRange")
+
+        path = (
+            settings.BASE_DIR
+            / APP_NAME
+            / "migrations"
+            / "data_migration_sources"
+            / "enrollment_tests.yaml"
+        )
+        with path.open(encoding="utf8") as fh:
+            data = yaml.safe_load(fh)
+
+        for block in data:
+            # id for Language is a string, so no need to get object from database to get its id
+            enrollment_test = EnrollmentTest.objects.create(language_id=block["language"])
+
+            enrollment_test.age_ranges.set(
+                AgeRange.objects.filter(
+                    type=AgeRangeType.STUDENT,
+                    age_from__gte=block["age_from"],
+                    age_to__lte=block["age_to"],
+                )
+            )
+
+            questions = block["questions"]
+
+            for item in questions:
+                question = EnrollmentTestQuestion.objects.create(
+                    enrollment_test=enrollment_test, text=item["text"]
+                )
+
+                # make sure exactly one option is marked as correct in each question
+                assert (
+                    len([o for o in item["options"] if o["is_correct"] is True]) == 1
+                ), f"Exactly 1 option has to be marked as correct for {item=}"
+
+                options = [
+                    EnrollmentTestQuestionOption(
+                        question=question,
+                        text=option_data["text"],
+                        is_correct=option_data["is_correct"],
+                    )
+                    for option_data in item["options"]
+                ] + [  # add "don't know" to each question
+                    EnrollmentTestQuestionOption(
+                        question=question,
+                        text="(I don't know 😕)",
+                        is_correct=False,
+                    )
+                ]
+                EnrollmentTestQuestionOption.objects.bulk_create(options)
```

Comment:
I think you can move this outside the for-loop, to be called only once. Also I'm not sure about question, but maybe it's also could be put into list for future bulk_create.

Reply:
Done for options (don't think can be done for questions)